### PR TITLE
Replaced .format{}/% with f-strings in core/tools/datetime.py

### PR DESCRIPTION
**Pull request by JMBurley**
Modified to python3 format strings
ref: https://github.com/pandas-dev/pandas/issues/29547

- [x] contributes to fstring corrections in  #29547
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
~~whatsnew entry~~ (will not add now. I think the full set of changes in #29547 should get a single update entry rather than a piecemeal one)

### CODE REVIEW

**Comment by JMBurley** [comment]
suggest labels clean & style for this pull request, & part of the 1.0 milestone.

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -485,27 +484,20 @@ def _adjust_to_origin(arg, origin, unit):
             or is_numeric_dtype(np.asarray(arg))
         ):
             raise ValueError(
-                "'{arg}' is not compatible with origin='{origin}'; "
-                "it must be numeric with a unit specified ".format(
-                    arg=arg, origin=origin
-                )
+                f"'{arg}' is not compatible with origin='{origin}'; "
+                "it must be numeric with a unit specified "
```
Comment (jbrockmendel) [inline pandas/core/tools/datetimes.py]: pls remove trailing space

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -810,8 +802,8 @@ def f(value):
     if len(req):
         raise ValueError(
             "to assemble mappings requires at least that "
-            "[year, month, day] be specified: [{required}] "
-            "is missing".format(required=",".join(req))
+            f"[year, month, day] be specified: [{','.join(req)}] "
```
Comment (jbrockmendel) [inline pandas/core/tools/datetimes.py]: pls do the join in a separate line

Comment (JMBurley) [reply pandas/core/tools/datetimes.py]: Done, see updated commit

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (JMBurley) [reply pandas/core/tools/datetimes.py]: Done, see updated commit

**Review by alimcmaster1 — APPROVED** [review]
LGTM

**Comment by jreback** [comment]
thanks @JMBurley

**Pull request merged**